### ISSUE TITLE
configcat.NewClient should not panic when SDK key is missing, should return an error instead

### ISSUE THREAD

**Issue by easyCZ** [configcat/go-sdk#37]
### Describe the bug
When `configcat.NewClient()` is invoked, it should not panic when the SDK Key provided is empty. Instead, it should return an error which can be checked by the caller who can then decide how to handle it. This is based on Golang best practice where panic should be reserved for fatal issues but not an error which can be checked.

### To reproduce
```golang
import "github.com/configcat/go-sdk/v6"

configcat.NewClient("") # panicks
```

### Expected behavior

`configcat.NewClient(..)` returns an exported error which can be compared against by the caller. For example 
```golang
client, err := configcat.NewClient("")
if err == configcat.NoSDKKeyProvidedError {
	# handle somehow
}
```

**Comment by easyCZ** [comment]
Digging through the SDK further, I've also noticed that other calls will also panic. These should also return an error, or return the default value in this case.

* `GetValueForUse`
* `GetValueAsyncForUser`
* `GetVariationIdForUser`
* `GetVariationIdAsyncForUser`

**Comment by z4kn4fein** [comment]
Hey, thanks for reporting this issue! 
We are releasing a new major version (v7) soon (in the next couple of days) from where we removed all those panics.

**Comment by easyCZ** [comment]
@z4kn4fein Thanks for letting me know. This is not urgent as I can work around it with a wrapper on my side where I check for these.

Happy to validate your v7 once you get around to rolling it out.

**Comment by z4kn4fein** [comment]
@easyCZ Just wanted to let you know that I released [v7](https://github.com/configcat/go-sdk/releases/tag/v7.0.0) a couple of minutes ago. 
If you have some time, could you please validate it? Many thanks!

**Comment by easyCZ** [comment]
Thanks. I should be able to give it a go later today and can get back to you with any feedback.

**Comment by easyCZ** [comment]
Hey @z4kn4fein, I've now upgraded my integration to v7.  It was largely painless and my test suite passed.

As some feedback on the library itself, see below:

* In my wrapper, I increment prometheus metrics when an experiment is evaluated. I record the experiment name and the resulting value. I'd find it useful to be able to have a hook (side effect only) on when a flag is evaluated so that I can add my metrics hook more easily.
* Similarly to the above, I also track metrics for downloads of the `config`. This is largely so that I have an idea about the latency and I can reason about any blocking behaviour in the application. The current design means I do this as a custom Transport/RoundTripper which increments the metric with latency and status code of the response. Having a hook for this would also be nice.

I'll have some more feedback once I've used more in anger :)
